invenio_app_rdm/records_ui/views/records.py

```diff
@@ -233,7 +237,7 @@ def record_detail(
         external_resources=get_external_resources(record_ui),
         user_avatar=avatar,
         record_owner_username=(
-            record_owner.username if record_owner is not None else None
+            record_owner.get("username", None) if record_owner != {} else None
```

Comment:
```suggestion
            record_owner.get("username")
```

Worst case, `record_owner` is an empty dict `{}`. It will never be None anymore.